utils.py

```diff
@@ -0,0 +1,29 @@
+
+class TrainState:
+    """Track number of steps, examples, and tokens processed"""
+
+    step: int = 0  # Steps in the current epoch
+    accum_step: int = 0  # Number of gradient accumulation steps
+    samples: int = 0  # total # of examples used
+    tokens: int = 0  # total # of tokens processed
+
+
+class SimpleLossCompute:
+    """A simple loss compute and train function."""
+
+    def __init__(self, generator, criterion):
```

Comment:
```suggestion
class SimpleLossCompute:
    """A simple loss compute and train function."""

    def __init__(self, generator: nn.Module, criterion: LabelSmoothing):
```

I know it's copied from the blogpost, but I just wanted to point out that this class name and doc string are really bad at describing what is the reason for this class to exist. 

Can you propose a better name for it? If not, this may indicate that we don't really need this class, and could be possible to handle loss calculation without a wrapper 🤔